### PR TITLE
feat: Adds BigQuery model extract jobs

### PR DESCRIPTION
**Pull request by amanda-tarafa**
One commit at a time.
The last two commits are minor cleanup and improvements identified when doing the main work. Happy to move those to another PR though.

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/BigQueryMLFixture.cs [hunk 1]
```diff
@@ -26,12 +24,19 @@ public class BigQueryMLFixture : CloudProjectFixtureBase, ICollectionFixture<Big
...
+        public StorageClient StorageClient { get; }
+
         public BigQueryMLFixture()
         {
             DatasetId = IdGenerator.FromDateTime(prefix: "testml_");
             ModelId = CreateModelId();
+            StorageClient = StorageClient.Create();
 
             CreateData();
+            StorageBucketName = IdGenerator.FromDateTime(prefix: "bigquerytestsml-");
```
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/BigQueryMLFixture.cs]: Please add this prefix to https://github.com/googleapis/google-cloud-dotnet/blob/master/apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.CleanTestData/Program.cs

Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/BigQueryMLFixture.cs]: (Or just use bigquerytests- as the prefix.)

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/ModelTests.cs [hunk 2]
```diff
@@ -154,5 +154,38 @@ public async Task GetModelAsync()
...
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var modelReference = client.GetModel(_fixture.DatasetId, _fixture.ModelId);
+            var destinationBucket = _fixture.StorageBucketName;
+            var destinationPrefix = _fixture.GenerateStoragePrefixName();
+            var destinationUri = $"gs://{destinationBucket}/{destinationPrefix}";
+
+            client.CreateModelExtractJob(modelReference.Reference, destinationUri).PollUntilCompleted().ThrowOnAnyError();
+
+            // We don't know the format of the extracted model,
+            // there are several files, so, let's check that we at least
```
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/ModelTests.cs:170]: Incomplete comment.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/ModelTests.cs [hunk 3]
```diff
@@ -154,5 +154,38 @@ public async Task GetModelAsync()
...
+            var modelReference = client.GetModel(_fixture.DatasetId, _fixture.ModelId);
+            var destinationBucket = _fixture.StorageBucketName;
+            var destinationPrefix = _fixture.GenerateStoragePrefixName();
+            var destinationUri = $"gs://{destinationBucket}/{destinationPrefix}";
+
+            var job = await client.CreateModelExtractJobAsync(modelReference.Reference, destinationUri);
+            job.PollUntilCompleted().ThrowOnAnyError();
+
+            // We don't know the format of the extracted model,
+            // there are several files, so, let's check that we at least
```
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/ModelTests.cs:187]: Ditto.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/ModelTests.cs [hunk 4]
```diff
@@ -154,5 +154,38 @@ public async Task GetModelAsync()
 
             Assert.Equal(modelId, model.Reference.ModelId);
         }
+
+        [Fact]
+        public void ModelExtractJob()
```
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/ModelTests.cs:159]: Rename this and next test to CreateModelExtractJob{Async}, to match the methods they're testing?

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/BigQueryFixtureBase.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+{
+    public abstract class BigQueryFixtureBase : CloudProjectFixtureBase
+    {
+        /// <summary>
+        /// A GCS bucket created for this fixture.
+        /// </summary>
+        public string StorageBucketName { get; }
+        public StorageClient StorageClient { get; }
+
+        public BigQueryFixtureBase(string bucketPrefix)
```
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/BigQueryFixtureBase.cs:28]: Could we just use bigquerytests- in all cases?

**Review by jskeet — APPROVED** [review]
Looks good to me - some nits, but that's all.

**Review by amanda-tarafa — COMMENTED** [review]
I've addresed your comments and will push shortly.
Let me know if you'd prefer one single fixture.

**Comment by jskeet** [comment]
Fine to keep it as different names for now. We can always revisit later (e.g. with some sort of atomicity guarantee) if we want.

**Pull request merged**